modes/play/render.js

```diff
@@ -80,6 +84,11 @@ class PlayModeRender {
     ctx.fillRect(screenX, screenY, width, height);
   }
 
+  drawImage(image, x, y, width, height) {
+    const ctx = this.game.ctx;
+    ctx.drawImage(image, x, y, width, height);
```

Comment:
Ah. This line is just printing the raw x, y coordinate from the object, and ignoring the camera offset. 

Notice the first line of `drawRect` where it transforms the input coordinates into `screenX, screenY` and passes those to `fillRect`. If you do the same transform for `drawImage`, the problem is fixed.